core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3623,9 +3639,16 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
   def foreachParDiscard[R, E, A](
     as: => Iterable[A]
   )(f: A => ZIO[R, E, Any])(implicit trace: Trace): ZIO[R, E, Unit] =
-    ZIO.parallelismWith {
-      case Some(n) => foreachParDiscard(n)(as)(f)
-      case None    => foreachParUnboundedDiscard(as)(f)
+    ZIO.succeed(as).flatMap { as =>
```

Comment:
I just wanted to avoid accidentally using `as` (the lazy `=> Iterable[A]`) as method arguments, but I think `suspendSucceed` is probably cleaner so I'll change this